src/xdsl/ir.py

```diff
@@ -196,7 +196,7 @@ def build(cls: Type[A], *args: Any) -> A:
 
 
 @dataclass(frozen=True)
-class Data(Generic[DataElement], Attribute):
+class Data(Generic[DataElement], Attribute, ABC):
```

Comment:
I don't get why this change is necessary.

Reply:
Oh, I just added it because I saw it missing. From what I understand, it improves error messages on IDE (and possibly at runtime).